sync_module/selective_sync/selective_sync.go

```diff
@@ -0,0 +1,167 @@
+	VisionService string `json:"vision_service"`
+}
+
+// Validate validates the config and returns implicit dependencies.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.DataManager == "" {
+		return nil, fmt.Errorf(`expected "data_manager" attribute in %q`, path)
+	}
+	if cfg.Camera == "" {
+		return nil, fmt.Errorf(`expected "camera" attribute in %q`, path)
+	}
+	if cfg.VisionService == "" {
+		return nil, fmt.Errorf(`expected "vision_service" attribute in %q`, path)
+	}
+
+	return []string{cfg.DataManager, cfg.Camera, cfg.VisionService}, nil
+}
+
+type visionSyncer struct {
+	resource.Named
+	actualDM      datamanager.Service
+	camera        camera.Camera
+	visionService vision.Service
+
+	cancelCtx  context.Context
+	cancelFunc func()
+	logger     golog.Logger
+}
+
+func newSelectiveSyncer(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (resource.Resource, error) {
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	v := &visionSyncer{
+		Named:      conf.ResourceName().AsNamed(),
+		logger:     logger,
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+	}
+	if err := v.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	v.startBackgroundProcess()
+	return v, nil
+}
+
+func (s *visionSyncer) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	dmConfig, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	s.actualDM, err = datamanager.FromDependencies(deps, dmConfig.DataManager)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get dm %v for visionSyncer", dmConfig.DataManager)
+	}
+	// If sync_disabled is false for the data manager, there are no guarantees that the selective sync
+	// behavior will function as intended. It is possible that data will be synced irregardless of the trigger.
+	s.logger.Warnw("Behavior is undetermined when scheduled sync is enabled and using selective sync", "data manager", dmConfig.DataManager)
+	s.camera, err = camera.FromDependencies(deps, dmConfig.Camera)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get camera %v for visionSyncer", dmConfig.VisionService)
+	}
+	s.visionService, err = vision.FromDependencies(deps, dmConfig.VisionService)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get vision service %v for visionSyncer", dmConfig.VisionService)
+	}
+
+	return nil
+}
+
+// DoCommand simply echos whatever was sent.
+func (s *visionSyncer) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+	return cmd, nil
+}
+
+func (s *visionSyncer) startBackgroundProcess() {
+	utils.PanicCapturingGo(func() {
+		stream, err := s.camera.Stream(s.cancelCtx)
+		defer utils.UncheckedErrorFunc(func() error {
+			return stream.Close(s.cancelCtx)
+		})
+		if err != nil {
+			s.logger.Error("could not get camera stream")
+			return
+		}
+		ticker := time.NewTicker(time.Second)
```

Comment:
Going back and forth about whether this should be in the config, but for the simplest possible example left this as a fixed value.

Reply:
Since this is an example, hardcoding it seems fine.